jax/_src/dispatch.py

```diff
@@ -326,7 +312,7 @@ def lower_xla_callable(fun: lu.WrappedFun, device, backend, name,
                       if i in kept_var_idx]
     del kept_const_idx
   else:
-    kept_var_idx = set(range(len(abstract_args)))
+    kept_var_idx = set(range(len(fun.in_type)))
```

Comment:
We need to keep all arguments whether implicit or explicit. Previously this was buggy in the presence of implicit arguments.